core/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java

```diff
@@ -122,4 +135,49 @@ public boolean equals(Object obj) {
         return super.equals(obj) &&
                 Objects.equals(format, other.format);
     }
+
+    protected static DocValueFormat wrapStringsInFormatter(final Map<Object, String> valuesAsStrings) {
+        return new DocValueFormat() {
```

Comment:
I have been trying to find an alternative to this but I didn't come up with anything better. Would be nice to be able to return the string values without going through the formatter for the client, but that is not a trivial change.

Shall we have two variants of this class, one for single value (no map then), and one for multiple values? Seems like most of the classes have a single value and don't need a map?